ui/src/app/components/workflows/workflow-form/dynamic-parts/string-with-suggestions-part/string-with-suggestions-part.component.spec.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+
+import { StringWithSuggestionsPartComponent } from './string-with-suggestions-part.component';
+import { Subject } from 'rxjs';
+import { WorkflowEntryModel, WorkflowEntryModelFactory } from '../../../../../models/workflowEntry.model';
+import { By } from '@angular/platform-browser';
+import { DebugElement, Predicate } from '@angular/core';
+import { FormsModule, NgForm } from '@angular/forms';
+import { PartValidationFactory } from '../../../../../models/workflowFormParts.model';
+
+describe('StringWithSuggestionsPartComponent', () => {
+  let fixture: ComponentFixture<StringWithSuggestionsPartComponent>;
+  let underTest: StringWithSuggestionsPartComponent;
+
+  const inputSelector: Predicate<DebugElement> = By.css('input[type="text"]');
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        declarations: [StringWithSuggestionsPartComponent],
+        imports: [FormsModule],
+        providers: [NgForm],
+      }).compileComponents();
+    }),
+  );
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(StringWithSuggestionsPartComponent);
+    underTest = fixture.componentInstance;
+  });
+
+  it('should create', () => {
+    expect(underTest).toBeTruthy();
+  });
+
+  describe('should set empty string on init when value is undefined or null', () => {
+    const parameters = [null, undefined];
+
+    parameters.forEach((parameter) => {
+      it(
+        'should pass with ' + parameter + ' value',
+        waitForAsync(() => {
+          const oldValue = parameter;
+          const newValue = '';
+          const options = ['first', 'scond'];
```

Comment:
Typo. Found nothing else to comment on :)

Reply:
Fixed